### ISSUE TITLE
[JENKINS-12063] Automatically update graphs when job finishes

### ISSUE THREAD

**Issue by jenkins-infra-bot** [timja-org/jenkins-gh-issues-poc-11-15#4974]
<p>You are on a job page, the job is in progress, regardless of whether there is compiler warnings graph (2+ jobs run before) or no graph (only one job run before), when the job finishes, the build status updates but not the graph.</p>

<p>Trend graphs are represented by static HTML in the moment. They will be updated only if the page is refreshed (see refresh interval). </p>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=moncreaff">moncreaff</a>, imported from: <a href="https://issues.jenkins.io/browse/JENKINS-12063" target="_blank">Automatically update graphs when job finishes</a></i></summary>
<i><ul>
<li><b>status</b>: Closed
<li><b>priority</b>: Minor
<li><b>component(s)</b>: analysis-core-plugin, core
<li><b>resolution</b>: Won't Fix
<li><b>resolved</b>: 2021-05-06T21:36:56+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 1
<li><b>imported</b>: 2025-11-16
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>You are on a job page, the job is in progress, regardless of whether there is compiler warnings graph (2+ jobs run before) or no graph (only one job run before), when the job finishes, the build status updates but not the graph.</p>

<p>Trend graphs are represented by static HTML in the moment. They will be updated only if the page is refreshed (see refresh interval). </p></pre>
</details>
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>moncreaff's <a href="https://issues.jenkins.io/browse/JENKINS-12063?focusedId=156526&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-156526">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Is the build history not gated by refresh delay?  Should it trigger a refresh regardless of the delay when job finishes, or are these to things completely independent of each other?</p>
</pre>
</details>

</details>
<p>Is the build history not gated by refresh delay?  Should it trigger a refresh regardless of the delay when job finishes, or are these to things completely independent of each other?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>moncreaff's <a href="https://issues.jenkins.io/browse/JENKINS-12063?focusedId=156527&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-156527">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>you may need to set your refresh rate to some large (multi 100) number of seconds to see what I mean</p>
</pre>
</details>

</details>
<p>you may need to set your refresh rate to some large (multi 100) number of seconds to see what I mean</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/drulli.png" title="drulli's avatar" /> drulli's <a href="https://issues.jenkins.io/browse/JENKINS-12063?focusedId=156959&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-156959">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Hmm, I'm not sure if I understand correctly (the comments and the title seem to be different): does the warnings graph update after the auto-refresh interval or not?</p>
</pre>
</details>

</details>
<p>Hmm, I'm not sure if I understand correctly (the comments and the title seem to be different): does the warnings graph update after the auto-refresh interval or not?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>moncreaff's <a href="https://issues.jenkins.io/browse/JENKINS-12063?focusedId=156980&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-156980">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Sorry for the confusion.  Yes it does update after the refresh interval.  I guess what I was trying to say is that regardless of the interval, if the build is being observed and the build progress changes to completed, should that trigger an immediate refresh?  </p>

<p>CONOPS is to set long refresh interval so pages aren't polled regardless of change, but when something interesting happens, page is auto updated then.</p>
</pre>
</details>

</details>
<p>Sorry for the confusion.  Yes it does update after the refresh interval.  I guess what I was trying to say is that regardless of the interval, if the build is being observed and the build progress changes to completed, should that trigger an immediate refresh?  </p>

<p>CONOPS is to set long refresh interval so pages aren't polled regardless of change, but when something interesting happens, page is auto updated then.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/drulli.png" title="drulli's avatar" /> drulli's <a href="https://issues.jenkins.io/browse/JENKINS-12063?focusedId=156981&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-156981">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Ok, now I understand. I think the build history is updated via AJAX calls, the build graphs are static HTML (PNG files). They are only updated upon a refresh. I'm not sure if this could be changed easily. Anyway, that part should be changed in core so that all plug-ins will benefit from such a change. </p>
</pre>
</details>

</details>
<p>Ok, now I understand. I think the build history is updated via AJAX calls, the build graphs are static HTML (PNG files). They are only updated upon a refresh. I'm not sure if this could be changed easily. Anyway, that part should be changed in core so that all plug-ins will benefit from such a change. </p>
